src/editor/app.js

```diff
@@ -22,12 +27,14 @@ function isCleanOrForced(generation) {
 UiDriver.registerEventHandler("C_CMD_MARK_CLEAN", function(msg, data, prevReturn) {
     forceDirty = false;
     changeGeneration = editor.changeGeneration(true);
-    UiDriver.sendMessage("J_EVT_CLEAN_CHANGED", isCleanOrForced(changeGeneration));
+    UiDriver.proxy.clean = isCleanOrForced(changeGeneration);
+    UiDriver.proxy.sendEditorEvent("J_EVT_CLEAN_CHANGED", isCleanOrForced(changeGeneration));
```

Comment:
Right now using `JsToCppProxy::getValue("some key")` does not work for clean events and maybe others. 

The problem is that lines such as this one send the `"J_EVT_CLEAN_CHANGED"` signal which is saved in the JsProxy's hash map. However, when querying whether the editor is clean, we generally use `"clean"` (see `Editor::isClean`).

As to fix that.. the most basic fix would be to just send the `"clean"` event alongside the  `"J_EVT_CLEAN_CHANGED"` event. But maybe it would be a good idea to split this into a `sendEditorEvent()` and a `setEditorData()` to distinguish between events and just setting data? Or maybe JsProxy should just react on ` `"J_EVT_CLEAN_CHANGED"`` events and set `"clean"` to the proper value? Then we wouldn't have to send any signals like `clean` or `language` anymore.

Right now it's a bit of a mess me thinks.